tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/OCLStamp.java

```diff
@@ -123,9 +123,14 @@ public boolean isCompatible(Constant constant) {
 
     @Override
     public boolean isCompatible(Stamp stamp) {
-        if (stamp instanceof OCLStamp && ((OCLStamp) stamp).oclKind == oclKind) {
-            return true;
-        }
+            if (stamp instanceof OCLStamp) {
```

Comment:
Shouldn't we have also this for the PTX backend?

Reply:
we do have it: https://github.com/beehive-lab/TornadoVM/pull/465/commits/faad244182a80fb793d5037be8a828ac8c77ab39